diff --git a/meson.build b/meson.build
index 0675a963..0e7788fa 100644
--- a/meson.build
+++ b/meson.build
@@ -276,6 +276,13 @@ if true
     src_files += files('src/modules/wlr/taskbar.cpp')
     man_files += files('man/waybar-wlr-taskbar.5.scd')
 endif
+    src_files += files('src/modules/wayfire/taskbar.cpp')
+    src_files += files('src/modules/sysmon/graph.cpp')
+    src_files += files('src/modules/hw/gauge.cpp')
+    src_files += files('src/modules/analog/clock.cpp')
+    src_files += files('src/modules/media/card.cpp')
+    src_files += files('src/modules/hw/bluetooth.cpp')
+    src_files += files('src/modules/cal/card.cpp')
 
 if wayland_protos.version().version_compare('>=1.39')
     add_project_arguments('-DHAVE_EXT_WORKSPACES', language: 'cpp')
diff --git a/src/factory.cpp b/src/factory.cpp
index 7828ce75..4a1bca8f 100644
--- a/src/factory.cpp
+++ b/src/factory.cpp
@@ -17,6 +17,13 @@
 #ifdef HAVE_WLR_TASKBAR
 #include "modules/wlr/taskbar.hpp"
 #endif
+#include "modules/wayfire/taskbar.hpp"
+#include "modules/sysmon/graph.hpp"
+#include "modules/hw/gauge.hpp"
+#include "modules/analog/clock.hpp"
+#include "modules/media/card.hpp"
+#include "modules/hw/bluetooth.hpp"
+#include "modules/cal/card.hpp"
 #ifdef HAVE_EXT_WORKSPACES
 #include "modules/ext/workspace_manager.hpp"
 #endif
@@ -178,6 +185,27 @@ waybar::AModule* waybar::Factory::makeModule(const std::string& name,
       return new waybar::modules::wlr::Taskbar(id, bar_, config_[name]);
     }
 #endif
+    if (ref == "wayfire/taskbar") {
+      return new waybar::modules::wayfire::Taskbar(id, bar_, config_[name]);
+    }
+    if (ref == "sysmon/graph") {
+      return new waybar::modules::sysmon::Graph(id, bar_, config_[name]);
+    }
+    if (ref == "hw/gauge") {
+      return new waybar::modules::hw::Gauge(id, bar_, config_[name]);
+    }
+    if (ref == "clock/analog") {
+      return new waybar::modules::analog::Clock(id, bar_, config_[name]);
+    }
+    if (ref == "media/card") {
+      return new waybar::modules::media::Card(id, bar_, config_[name]);
+    }
+    if (ref == "hw/bluetooth") {
+      return new waybar::modules::hw::Bluetooth(id, bar_, config_[name]);
+    }
+    if (ref == "cal/card") {
+      return new waybar::modules::cal::Card(id, bar_, config_[name]);
+    }
 #ifdef HAVE_EXT_WORKSPACES
     if (ref == "ext/workspaces") {
       return new waybar::modules::ext::WorkspaceManager(id, bar_, config_[name]);
